### ISSUE TITLE
Git overwrites local changes

### ISSUE THREAD

**Issue by TomasHubelbauer** [isomorphic-git/isomorphic-git#1046]
I am using the Node vesion, specifically version 0.78.4. I have noticed a behavior which doesn't match that of canonical Git.

I know there has been a few tickets about the behavior of `pull` and 3 way merges etc., but I don't think what I've found has been discussed yet - or was it?

This is the repro:
https://github.com/TomasHubelbauer/isogit-pull

1. Clone a repository
2. Go in and make a local change to a file, do not stage it
3. Go in and make a remote change to the same file, commit it
4. Pull the remote changes

Isomorphic-Git will discard the local changes to the file. I tried with and without the `fastForwardOnly` flag.

Canonical Git will error with conflicts encountered.

I am sure this must be known already so sorry in advance for filing dupes! Searching the issues for *pull conflicts* results in a lot of dependat issues and I don't know how to filter the bot out. :-)

**Comment by jcubic** [comment]
I don't think it was ever disscussed. To clarify the behavior of canonical git is to show warning that pull can't be done and changes must be commited or stashed. I think that isomorphic git should throw error in this case.

**Comment by TomasHubelbauer** [comment]
I agree that an error with a flag to disable it would be ideal, current behavior is basically data loss by default, could burn some people. :-)

**Comment by jcubic** [comment]
If git pull overwrite the changes there at least need to be a way to check if pull will do that. Maybe there is need to do fetch and some check before the merge. That will be completly new pull function, that is correct, maybe the pull should do that if check fail it do revert the changes and never do merge.

**Comment by wmhilton** [comment]
Yup! Yup. You could technically avoid that by doing a `fetch` and then a `fastCheckout` (or by doing a `pull({ ..., fast: true })`. This is one of the things that (should) be fixed in version 1.0! 🎉
